packages/form-system/stories/04-interactionStates.stories.mdx

```diff
@@ -0,0 +1,185 @@
+import { Story, Meta, html } from '@open-wc/demoing-storybook';
+import { render } from '@lion/core';
+import '@lion/input/lion-input.js';
+import { Validator } from '@lion/validate';
+import './helper-wc/h-output.js';
+
+<Meta title="Form Fundaments/Interaction States" />
+
+# Interaction States
+
+`InteractionStateMixin` saves meta information about interaction states.
+It allows for creating advanced UX scenarios.
+
+Examples of such scenarios already in our fields:
+
+- Show the validation message of an input only after the user has blurred the input field
+- Hide the validation message when an invalid value becomes valid
+
+Something our subclassers can implement:
+
+- Show a red border around the input right after the input became invalid
+
+The meta information that InteractionStateMixin collects, is stored in the Boolean properties on our fields:
+
+- `touched`, the user blurred the field
+- `dirty`, the value in the field has changed
+- `prefilled`, a prepopulated field is not empty
+
+> You can listen to the events `touched-changed` and `dirty-changed`.
+
+<Story name="Interaction States">
+  {html`
+    <lion-input
+      label="Interaction States"
+      help-text="Interact with this field to see how dirty, touched and prefilled change"
+      .modelValue="${'myValue'}"
+    ></lion-input>
+    <h-output .show="${['touched', 'dirty', 'prefilled', 'focused', 'submitted']}"></h-output>
+  `}
+</Story>
+
+```html
+<lion-input
+  label="Interaction States"
+  help-text="Interact with this field to see how dirty, touched and prefilled change"
+  .modelValue="${'myValue'}"
+></lion-input>
+```
+
+## Advanced use cases
+
+### Overriding interaction states
+
+When creating an extension of LionField or LionInput, it can be needed to override the way prefilled values are 'computed'.
```

Comment:
```suggestion
When creating an extension of LionField or LionInput, it can be useful to override the way prefilled values are 'computed'.
```

Reply:
In case of the ChoiceInputMixin, it's actually needed ;)